code/game/machinery/landline.dm

```diff
@@ -0,0 +1,393 @@
+		for (var/obj/machinery/message_server/MS in message_servers)
+			if(MS.landlines_functioning())
+				calling.calling = null
+				calling = null
+				break
+	/* for(var/obj/machinery/computer/message_monitor/MM in message_monitors)
+		MM.updateUsrDialog()
+		message_admins("[MM] updated") */
+	//TODO make this actually update the switchboard window
+	
+	user.visible_message("<span class='notice'>[user] puts \the [O] onto \the [src.attached_to].</span>")
+	var/obj/item/telephone/P = O
+	playsound(source=O, soundin=P.pickup_sound, vol=100, vary=TRUE, channel=0)
+	phone = O
+	O.forceMove(src)
+	attached_to.overlays.Add(phone_overlay)
+	
+/obj/landline/red
+	overlay_icon = 'icons/obj/items.dmi'
+	overlay_iconstate = "red_phone_handset"
+	
+	
+/obj/item/telephone
+	name = "telephone"
+	icon = 'icons/obj/terminals.dmi'
+	icon_state = "phone"
+	inhand_states = list("left_hand" = 'icons/mob/in-hand/left/misc_tools.dmi', "right_hand" = 'icons/mob/in-hand/right/misc_tools.dmi')
+	item_state = "rpb_phone"	
+	flags = HEAR | FPRINT
+	var/mic_range = 3
+	var/speaker_range = 3
+	var/soundloops_before_exploding = 3
+	var/obj/landline/linked_landline = null
+	var/datum/speech/lastmsg
+	var/pickup_sound = 'sound/items/telephone_pickup.ogg'
+	var/ringtone = 'sound/items/telephone_ring.ogg'
+	var/end_call_sound = 'sound/items/telephone_end_call_440hz.mp3'
+	var/dial_sound = 'sound/items/telephone_dial_440hz.mp3'
+	var/clowned = FALSE 
+	var/broken = FALSE
+		
+/obj/item/telephone/attackby(var/obj/item/weapon/O as obj, var/mob/user as mob)
+	if(istype(O,/obj/item/toy/crayon/rainbow) && clowned == FALSE)
+		to_chat(user, "<span class = 'notice'>You begin modifying \the [src].</span>")
+		if(do_after(user, src, 4 SECONDS))
+			to_chat(user, "<span class = 'notice'>You finish modifying \the [src]!</span>")
+			honkify()
+			clowned = TRUE
+			
+/obj/item/telephone/proc/honkify()
+	name = "Bananaphone"
+	icon = 'icons/obj/hydroponics/banana.dmi'
+	icon_state = "produce"
+	pickup_sound = 'sound/items/bananaphone_pickup.ogg'
+	ringtone = 'sound/items/bananaphone_ring.ogg'
+	if(linked_landline)
+		linked_landline.phone_overlay = image(icon = linked_landline.overlay_icon, icon_state = "phone_overlay_banana")
+	update_icon()
+	
+/obj/item/telephone/proc/make_cord()
+	if(!linked_landline)
+		return
+	linked_landline.delete_cord()
+	linked_landline.make_cord()
+	
+/obj/item/telephone/pickup(var/mob/user)
+	..()
+	make_cord()
+	user?.register_event(/event/after_move, src, /obj/item/telephone/proc/make_cord)	
+	
+/obj/item/telephone/dropped(var/mob/user)
+	..()
+	make_cord()
+	user?.unregister_event(/event/after_move, src, /obj/item/telephone/proc/make_cord)	
+
+/obj/item/telephone/Move(NewLoc, Dir = 0, step_x = 0, step_y = 0, glide_size_override = 0)
+	//TODO make this work while in lockers and backpacks and such
+	..()
+	if(!linked_landline)
+		return
+	linked_landline.delete_cord()
+	linked_landline.make_cord()
+
+/obj/item/telephone/proc/explode()

```

Comment:
Why does this have a snowflaked explosion with a stun?

Reply:
because i had to break the infinite loop somehow and anonymous suggested this